lsp/src/Language/LSP/Server/Core.hs

```diff
@@ -49,23 +49,47 @@ import           Data.Text ( Text )
 import qualified Data.UUID as UUID
 import Language.LSP.Protocol.Types
 import Language.LSP.Protocol.Message
-import qualified Language.LSP.Protocol.Types as J
-import qualified Language.LSP.Protocol.Lens as J
-import qualified Language.LSP.Protocol.Message as J
+import qualified Language.LSP.Protocol.Types as L
+import qualified Language.LSP.Protocol.Lens as L
+import qualified Language.LSP.Protocol.Message as L
 import           Language.LSP.Protocol.Utils.SMethodMap (SMethodMap)
 import qualified Language.LSP.Protocol.Utils.SMethodMap as SMethodMap
 import           Language.LSP.VFS
 import           Language.LSP.Diagnostics
 import           System.Random hiding (next)
 import           Control.Monad.Trans.Identity
 import           Control.Monad.Catch (MonadMask, MonadCatch, MonadThrow)
+import Prettyprinter
 
 -- ---------------------------------------------------------------------
 {-# ANN module ("HLint: ignore Eta reduce"         :: String) #-}
 {-# ANN module ("HLint: ignore Redundant do"       :: String) #-}
 {-# ANN module ("HLint: ignore Reduce duplication" :: String) #-}
 -- ---------------------------------------------------------------------
 
+data LspCoreLog =
+  -- TODO: arguably it would be nicer to have the config object itself in there, but
+  -- then we're going to need 'Pretty config' constraints everywhere
+  NewConfig J.Value
+  | ConfigurationParseError J.Value T.Text
+  | ConfigurationNotSupported
+  | BadConfigurationResponse ResponseError
+  | WrongConfigSections [J.Value]
+  deriving Show
+
+instance Pretty LspCoreLog where
+  pretty (NewConfig config) = "LSP: set new config:" <+> viaShow config
+  pretty (ConfigurationNotSupported) = "LSP: not requesting configuration since the client does not support workspace/configuration"
+  pretty (ConfigurationParseError settings err) =
+    vsep [
+      "LSP: configuration parse error:"
+      , pretty err
+      , "when parsing"
```

Comment:
```suggestion
      , ", when parsing: "
```

Reply:
`vsep` inserts spaces between the items if they're formatted on one line